python/plugins/processing/algs/gdal/rearrange_bands.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    translate.py
+    ---------------------
+    Date                 : August 2018
+    Copyright            : (C) 2018 by Mathieu Pellerin
+    Email                : nirvn dot asia at gmail dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Mathieu Pellerin'
+__date__ = 'August 2018'
+__copyright__ = '(C) 2018, Mathieu Pellerin'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+import os
+import re
+
+from qgis.PyQt.QtGui import QIcon
+
+from qgis.core import (QgsRasterFileWriter,
+                       QgsProcessingException,
+                       QgsProcessingParameterEnum,
+                       QgsProcessingParameterDefinition,
+                       QgsProcessingParameterRasterLayer,
+                       QgsProcessingParameterBand,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterRasterDestination)
+from processing.algs.gdal.GdalAlgorithm import GdalAlgorithm
+from processing.algs.gdal.GdalUtils import GdalUtils
+
+pluginPath = os.path.split(os.path.split(os.path.dirname(__file__))[0])[0]
+
+
+class rearrange_bands(GdalAlgorithm):
```

Comment:
Don't forget unit tests for this algorithm (just go straight for the GDAL specific command tests, don't worry about standard processing tests here)